check-vmware/test_checks.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+import random
+import yaml
+
+from pyVmomi import vim
+from vmware_checks import CHECKS
+from vmware_logconf import get_logger
+from wrapanapi.systems.virtualcenter import VMWareSystem
+
+
+@pytest.fixture(scope="session")
+def provider_data():
+    try:
+        with open("conf/cfme-qe-yamls/complete/cfme_data.yaml", "r") as stream:
+            cfme_data = yaml.safe_load(stream)
+    except IOError:
+        pytest.fail("CFME Data YAML file not found, no provider data available.")
+    else:
+        providers = cfme_data.get("management_systems")
+        vsphere_providers = {
+            key: providers[key] for key in list(providers.keys())
+            if providers[key].get("type") == "virtualcenter"
+        }
+        yield vsphere_providers.get(random.choice(list(vsphere_providers.keys())))
+
+@pytest.fixture(scope="session")
+def credentials(provider_data):
+    try:
+        with open("conf/cfme-qe-yamls/complete/credentials.yaml", "r") as stream:
+            creds = yaml.safe_load(stream)
+            yield creds.get(provider_data.get("credentials"))
+    except IOError:
+        pytest.fail("Credential YAML file not found or not decrypted!")
+
+
+@pytest.mark.parametrize("measurement", list(CHECKS.keys()))
+def test_checks(provider_data, credentials, measurement):
+    # get the necessary config
+    if measurement == "system_ping_vms":
+        pytest.skip("Measurement {} is unused, so skipping".format(measurement))
+    measure_func = CHECKS[measurement]
+    hostname, credential_key = provider_data.get("hostname"), provider_data.get("credentials")
```

Comment:
Can you explain usage of `credential_key` please?

Reply:
`credential_key` is passed to `credentials` to get `username` and password

Reply:
Right, but I don't see this being used anywhere. Hence asked. Can you please check.

Reply:
Ah yep, thanks, that was some old leftover code.